-mfpmath=sse is not added in 32bits build

Hi,

-mfpmath=sse is not added in 32bits build, and from a comment in Jolt.cmake this seems to be a mistake

https://github.com/jrouwe/JoltPhysics/blob/4cd52055e09160affcafa557b39520331bf0d034/Jolt/Jolt.cmake#L647

This block is still part of the 64bits check, same for EMIT_X86_INSTRUCTION_SET_DEFINITIONS.

Also, shouldn't -msse2 be added too in 32bits build? -mfpmath=sse is not enough

Hello,

The errors in the referenced ticket don't seem to be related to enabling sse:

```
/opt/homebrew/Cellar/mingw-w64/12.0.0/toolchain-i686/lib/gcc/i686-w64-mingw32/14.1.0/include/xmmintrin.h: In member function 'float JPH::Vec3::GetX() const':
/opt/homebrew/Cellar/mingw-w64/12.0.0/toolchain-i686/lib/gcc/i686-w64-mingw32/14.1.0/include/xmmintrin.h:973:1: error: inlining failed in call to 'always_inline' 'float _mm_cvtss_f32(__m128)': target specific option mismatch
  973 | _mm_cvtss_f32 (__m128 __A)
      | ^~~~~~~~~~~~~
```

It's a warning about inlining which I already fixed in 4a08d8d4e1ce7e87fdb16b0708a54a15f9a5379e (which is not in an official release yet).

The `-mfpmath=sse` flag seems to be sufficient for GCC to start creating SSE instructions. I've created a sample in [godbolt](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:6,endLineNumber:1,positionColumn:6,positionLineNumber:1,selectionStartColumn:6,selectionStartLineNumber:1,startColumn:6,startLineNumber:1),source:'float+function()%0A%7B%0A++++float+x+%3D+1.0f%3B%0A++++float+y+%3D+2.0f%3B%0A++++return+x+*+y%3B%0A%7D'),l:'5',n:'1',o:'C%2B%2B+source+%231',t:'0')),k:37.184204071621295,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:g141,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'-m32+-mfpmath%3Dsse',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+gcc+14.1+(Editor+%231)',t:'0')),k:29.482462595045376,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compilerName:'x86-64+clang+(trunk)',editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+x86-64+gcc+14.1+(Compiler+%231)',t:'0')),k:33.33333333333333,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4). You can see that the generated assembly uses SSE instructions. If you remove the `-mfpmath=sse` flag, or replace it with `-msse`, it starts creating old style x87 instructions.

If this is the 64-bit section you're referring to:

https://github.com/jrouwe/JoltPhysics/blob/d3c4e3058a98a4b6ca03076e8555ff4fb17899a6/Jolt/Jolt.cmake#L619

then what it is checking is `CMAKE_SYSTEM_PROCESSOR`, which in my experience for x86 is [the processor that is compiling the code, not what architecture it is targetting](https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_PROCESSOR.html). I'm hoping you're not compiling on a 32-bit OS, but if you are, or if I'm wrong about the value, we can add an extra check for that.

--- a/Jolt/Jolt.cmake
+++ b/Jolt/Jolt.cmake
@@ -616,8 +616,12 @@ else()
 		# XCode builds for multiple architectures, we can't set global flags
 	elseif (CROSS_COMPILE_ARM OR CMAKE_OSX_ARCHITECTURES MATCHES "arm64" OR "${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "aarch64")
 		# ARM64 uses no special commandline flags
-	elseif ("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86_64" OR "${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "AMD64")
-		# x64
+	elseif ("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86_64" OR "${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "AMD64" OR "${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86" OR "${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "i386")
+		# x86 and x86_64
+  		# On 32-bit builds we need to default to using SSE instructions, the x87 FPU instructions have higher intermediate precision
+		# which will cause problems in the collision detection code (the effect is similar to leaving FMA on, search for
+		# JPH_PRECISE_MATH_ON for the locations where this is a problem).
+  
 		if (USE_AVX512)
 			target_compile_options(Jolt PUBLIC -mavx512f -mavx512vl -mavx512dq -mavx2 -mbmi -mpopcnt -mlzcnt -mf16c)
 		elseif (USE_AVX2)
@@ -644,9 +648,6 @@ else()
 			target_compile_options(Jolt PUBLIC -mfma)
 		endif()
 
-		# On 32-bit builds we need to default to using SSE instructions, the x87 FPU instructions have higher intermediate precision
-		# which will cause problems in the collision detection code (the effect is similar to leaving FMA on, search for
-		# JPH_PRECISE_MATH_ON for the locations where this is a problem).
 		if (NOT MSVC)
 			target_compile_options(Jolt PUBLIC -mfpmath=sse)
 		endif()

